File: quvault/core/src/main/scala/net/ellisw/quvault/core/QuContainer.scala
package net.ellisw.quvault.core;

import scala.collection.mutable.ArrayBuffer

class QuContainer(parent: QuObject) extends QuObject(parent) {
  val children = new ArrayBuffer[QuObject]();

  def createContainer(): QuContainer = {
    val o = new QuContainer(this)
    children += o
    o 
  }

  def createTextHtml(html: String): QuTextHtml = {
    val o = new QuTextHtml(html, this)
    children += o
    o 
  }
/*
	public QuProblem createProblem() {
		QuProblem o = new QuProblem(this);
		children.add(o);
		return o; 
	}
	
	public QuQuestionMathlib createQuestionMathlib() {
		QuQuestionMathlib o = new QuQuestionMathlib(this);
		children.add(o);
		return o; 
	}
	
	public QuText createText() {
		QuText o = new QuText(this);
		children.add(o);
		return o; 
	}
 */
}

File: quvault/vault-core-java/src/main/java/net/ellisw/quvault/vault/QuContainer.java
package net.ellisw.quvault.vault;

import java.util.ArrayList;
import java.util.List;

import org.dom4j.Element;

public class QuContainer extends QuObject {
	private List<QuObject> children = new ArrayList<QuObject>();
	
	public List<QuObject> getChildren() { return children; }
	
	
	public QuContainer(QuObject parent) {
		super(parent);
	}
	
	public QuContainer createContainer() {
		QuContainer o = new QuContainer(this);
		children.add(o);
		return o; 
	}
	
	public QuProblem createProblem() {
		QuProblem o = new QuProblem(this);
		children.add(o);
		return o;
	}
	
	public QuProblemElec createProblemElec() {
		QuProblemElec o = new QuProblemElec(this);
		children.add(o);
		return o;
	}
	
	public QuQuestionMathlib createQuestionMathlib() {
		QuQuestionMathlib o = new QuQuestionMathlib(this);
		children.add(o);
		return o; 
	}
	
	public QuText createText() {
		QuText o = new QuText(this);
		children.add(o);
		return o; 
	}

	@Override
	public String getXmlTagName() {
		return "container";
	}

	@Override
	public void fillElement(Element elem) {
		super.fillElement(elem);
		
		for (QuObject child : children) {
			Element elemChild = child.getXml();
			elem.add(elemChild);
		}
	}
}
